cluster-autoscaler/provisioningrequest/provreqclient/client.go

```diff
@@ -41,15 +41,21 @@ const (
 	provisioningRequestClientCallTimeout = 4 * time.Second
 )
 
-// ProvisioningRequestClient represents client for v1beta1 ProvReq CRD.
-type ProvisioningRequestClient struct {
+// ProvisioningRequestClient is an interface with methods for v1beta1 ProvReq CRD client.
+type ProvisioningRequestClient interface {
```

Comment:
Why make this an interface? Usually interfaces should be defined in consumer packages (like they were before this PR), not the implementer package. See e.g. https://medium.com/@mbinjamil/using-interfaces-in-go-the-right-way-99384bc69d39 for an explanation why